src/main/java/appeng/util/inv/AdaptorList.java

```diff
@@ -41,6 +41,12 @@ public AdaptorList( final List<ItemStack> s )
 	}
 
 	@Override
+	public boolean hasSlots()
+	{
+		return i.size() > 0;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1155)